### DISCUSSION TITLE
Unable to use log() in my code facing linker error

### DISCUSSION THREAD

**Discussion by aravinth-kalai** [zephyrproject-rtos/zephyr#61553] [Q&A] [answered]
Hello Everyone !

I'm new to zephyr, 
I'm doing some Logarithmic calculation in my code, so that I just included math.h file and added math library in  Cmakelists, when I try to build my code , I'm facing linker error 
I'm unable to share full code , but I'm sharing the relevant code area
 `#include<math.h

void main()
{
log(22);
while(1)
{
}
}
`
`CMakeLists.txt
....
....
target_link_libraries(app PRIVATE m)
`

I'm getting this linker error:

`arm-gnu-toolchain-12.3.rel1-mingw-w64-i686-arm-none-eabi/bin/../lib/gcc/arm-none-eabi/12.3.1/thumb/v8-m.main/nofp\libm.a(libm_a-w_log.o): in function `log':`
w_log.c:(.text.log+0x16): undefined reference to ``__aeabi_dcmpun`

`arm-gn~1.rel/bin/../lib/gcc/arm-none-eabi/12.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: w_log.c:(.text.log+0x38): undefined reference to ``__errno``

`newfol~1/arm-gn~1.rel/bin/../lib/gcc/arm-none-eabi/12.3.1/../../../../arm-none-eabi/bin/ld.bfd.exe: w_log.c:(.text.log+0x4a): undefined reference to ``__errno'`

Board: lpcxpresso55s16
Zephyr version : 3.3.99
ZEPHYR_TOOLCHAIN_VARIANT : gnuarmemb

Please help me to fix this issue , if need more clarification please let me know,
Thanks in advance !

**Comment by yashi** [comment, marked as answer]
That's not how you do it in Zephyr.

apply the patch bellow and run the following command.  This means that
- You are using [Picolibc](https://docs.zephyrproject.org/latest/develop/languages/c/picolibc.html) instead of [minimal libc](https://docs.zephyrproject.org/latest/develop/languages/c/minimal_libc.html): `CONFIG_PICOLIBC=y`
- You tell you want to print float by `CONFIG_PICOLIBC_IO_FLOAT=y`

You can use interactive UI if you want: https://docs.zephyrproject.org/latest/build/kconfig/menuconfig.html

```
rm -rf build && west build -b qemu_cortex_m3 samples/hello_world -t run
```

```diff
diff --git a/samples/hello_world/prj.conf b/samples/hello_world/prj.conf
index b2a4ba59104..6bea429b0f3 100644
--- a/samples/hello_world/prj.conf
+++ b/samples/hello_world/prj.conf
@@ -1 +1,2 @@
-# nothing here
+CONFIG_PICOLIBC=y
+CONFIG_PICOLIBC_IO_FLOAT=y
diff --git a/samples/hello_world/src/main.c b/samples/hello_world/src/main.c
index bc7a4c1d716..362b433b353 100644
--- a/samples/hello_world/src/main.c
+++ b/samples/hello_world/src/main.c
@@ -5,9 +5,10 @@
  */
 
 #include <zephyr/kernel.h>
+#include <math.h>
 
 int main(void)
 {
-       printk("Hello World! %s\n", CONFIG_BOARD);
+       printk("Hello World! %lf\n", log(22));
        return 0;
 }
```

To build for your board,
```
rm -rf build && west build -b lpcxpresso55s16 samples/hello_world
```

BTW, please use triple backticks to encode your code.
````
```
your code here
```
````

**Reply by aravinth-kalai** [reply]
Thank for your reply @yashi , 

Sorry for late reply, 

This config is actually fixed my issue , but one information I should share that, this fix will work only if we build with SDK not with gnuarm tool chain, we must have zephyr SDK tool chain to use this config.
